chain-signatures/node/src/protocol/signature.rs

```diff
@@ -80,32 +125,33 @@ impl SignQueue {
             me,
             sign_rx,
             my_requests: VecDeque::new(),
-            other_requests: HashMap::new(),
+            requests: HashMap::new(),
             failed_requests: VecDeque::new(),
+            pending: HashMap::new(),
         }
     }
 
     pub fn len_mine(&self) -> usize {
-        self.my_requests.len()
+        self.requests
+            .values()
+            .filter(|r| r.proposer == self.me)
+            .count()
     }
 
     pub fn is_empty_mine(&self) -> bool {
         self.len_mine() == 0
     }
 
     pub fn len(&self) -> usize {
-        self.my_requests.len() + self.other_requests.len()
+        self.requests.len()
```

Comment:
`len()` and how it's used in `is_empt()` is no longer 100% obvious, as we are not counting pending requests. Did you check that callers are okay with that? And should we maybe add comments like "length of running requests, excluding pending requests" to the two methods?